lib/proto_s3/proto_s3_client.c

```diff
@@ -481,9 +482,12 @@ callback_head(void * cookie, uint8_t * buf, size_t buflen)
 			lens = len;
 	}
 
+	/* Sanity check. */
+	assert((status == 0) || ((status >= 100) && (status <= 599)));
```

Comment:
Absolutely not.  You should never `assert` what you received over the wire -- that would make it possible for the server to crash us.

This should be checked the same way as we check if a response is valid: `if (...) BAD("HEAD", "invalid HTTP status")`.